src/Parser.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace CSVParser;
+
+class Parser {
+
+    /*
+     * @var $path the file path to the csv file to be parsed.
+     * @var $headers The column headers for the CSV file that give context to the column values.
+     * @var $delimiter the delimiter value that separates column values for this CSV file.
+     */
+
+    private $path;
+    private $headers = [];
+    private $delimiter;
+
+    function __construct ($path, $headers = [], $delimiter = ',') {
+
+        /*
+         * Check that the path does exist or throws an exception
+         * Checks that the delimiter is a , | or /t
+         * Fills headers, if it's empty grabs first line from CSV instead
+         */
+
+        if(!file_exists($path)) throw new Exception("{$path} is not a valid file.");

```

Comment:
Again totally stylistic
```php
$this->path = $path;
if (!file_exists($path))
    throw new Exception("{$path} is not a valid file.")
```

I might also change the message to say "{$path} does not exist."